lib/kubeclient/common.rb

```diff
@@ -121,16 +123,21 @@ def discovery_needed?(method_sym)
 
     def handle_exception
       yield
-    rescue RestClient::Exception => e
+    rescue Faraday::Error => e
+      err_message = get_error_message(e)
+      response_code = e.response ? (e.response[:status] || e.response&.env&.status) : nil
+      error_klass = response_code == 404 ? ResourceNotFoundError : HttpError
```

Comment:
FYI imo better readability:
```suggestion
      error_klass = (response_code == 404 ? ResourceNotFoundError : HttpError)
```